### ISSUE TITLE
Helper url_for doesn't generate correct i18n URL for entirely localized templates

### ISSUE THREAD

**Issue by phvega** [middleman/middleman#2592]
## Expected behavior and actual behavior

With a Middleman config with i18n plugin activated as follows:

```rb
activate :i18n, :langs => [:en, :es], :mount_at_root => false
```
I have set up two localized pages, one of them uses [fully localized template](https://middlemanapp.com/advanced/localization/#localizing-entire-templates), the other doesn't:

- `source/localizable/path1.html.erb`
- `source/localizable/path2.en.html.erb`
- `source/localizable/path2.es.html.erb`

I want to get the reference to those pages using `url_for` helper from other pages, so I wrote this `source/localizable/index.html.erb`:

```erb
<h3><%= t(:home) %></h3>

<a href="<%= url_for '/path1.html' %>">Working path</a><br>
<a href="<%= url_for '/path2.html' %>">Wrong path</a>
```

However, the output of `/en/index.html` after building it is:

```html
<h3>Home</h3>

<a href="/en/path1.html">Working path</a><br> # Correct
<a href="/path2.html">Wrong path</a> # Should be /en/path2.html
```

As you can see, it is returning the correct reference to `path1.html`, but shows `/path2.html` for the second link instead of `/en/path2.html` as it should be.

In addition, if I change "path2" template filename from `path2.en.html.erb` to `path2.html.erb`, it generates the correct reference, i.e., "/en/path2.html"

## Steps to reproduce the problem (from a clean middleman installation)

I have uploaded this [test repository](https://github.com/phvega/url_for-test)

## Additional information

- Ruby version: ruby 2.6.8p205 (2021-07-07 revision 67951) [x86_64-linux]
- Middleman version: 4.4.3
- OS version: Ubuntu 20.04.1

**Comment by markets** [comment]
@tdreyno I think we can close this one, fixed by #2604

**Comment by guillerodriguez** [comment]
I confirm that the issue is fixed.
